Community Toolbox/scripts/tests_ArrayPopRandomTests/tests_ArrayPopRandomTests.gml

```diff
@@ -0,0 +1,157 @@
+// general note: since array_pop_random is non-deterministic
+// the tests might technically pass even if there is an error by getting "lucky" with the RNG state;
+// what should be guaranteed, is that as long as the function is implemented correctly, the tests pass every time
+
+function ArrayPopRandomTests(_run, _method) : VerrificMethodTest(_run, _method) constructor {
+    static test_subject = "array_pop_random";
+    
+    static should_return_undefined_for_empty_array = function() {
+        given_array([]);
+        when_array_pop_random_runs();
+        then_result().should_be_undefined();
+        then_source_array_length().should_be(0);
+    }
+    
+    static should_pop_value_from_single_item_array = function() {
+        given_array([42]);
+        when_array_pop_random_runs();
+        then_result().should_be(42);
+        then_source_array_length().should_be(0);
+    }
+    
+    static should_pop_value_from_multi_item_array = function() {
+        given_array([100, 300, 500, 700, 900]);
+        when_array_pop_random_runs();
+        then_result().should_be_one_of([100, 300, 500, 700, 900]);
```

Comment:
Does Verrific have a way to check that an element is NOT in an array? It might be useful to add an additional assertion of the form
```
the_randomly_chosen_value.should_not_be_one_of(the_original_array);
```
if such an assertion exists.

Reply:
There is no specific assertion for that (yet), but there's a general-purpose `assert_that(condition, onfailure)` assertion which is generally a go-to way to handle non-trivial assertions in the absence of built-in ones.

Thus, I added the extra `then_source_array_should_not_have_result()` checks to the tests where applicable.

(technically it doesn't have to be true if the array to pop elements from has duplicates, but neither test scenario has duplicates, so this check makes sense in context)